### PR TITLE
Remove Much of the HTTP API Surface Area

### PR DESCRIPTION
**Pull request by arschles**
In accordance with https://github.com/kubernetes-incubator/service-catalog/issues/54, this patch attempts to only remove the API surface area. It won’t refactor any code.

Fixes https://github.com/kubernetes-incubator/service-catalog/issues/54

### CODE REVIEW

File: controller/server/server.go [hunk 1]
```diff
@@ -46,46 +43,23 @@ func CreateServer(serviceStorage ServiceStorage, port int, w *watch.Watcher) (*S
...
+	router.HandleFunc("/healthz", healthZHandler).Methods("GET")
 
 	port := strconv.Itoa(s.port)
 	log.Println("Server started on port " + port)
 	err := http.ListenAndServe(":"+port, nil)
 	log.Println(err.Error())
 }
+
+func healthZHandler(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
```
Comment (vaikas-google) [inline controller/server/server.go]: Consider adding a TODO here to do some actual health checking?

Comment (arschles) [inline controller/server/server.go]: @vaikas-google I'll add an issue for it instead. cool with you?

Comment (arschles) [inline controller/server/server.go]: https://github.com/kubernetes-incubator/service-catalog/issues/85

**Comment by krancour** [comment]
LGTM

File: controller/server/server.go [hunk 2]
```diff
@@ -46,46 +43,23 @@ func CreateServer(serviceStorage ServiceStorage, port int, w *watch.Watcher) (*S
...
-		k8sHandler:  k8sHandler,
+		controller: c,
+		port:       port,
+		k8sHandler: k8sHandler,
 	}, nil
 }
 
 // Start starts the server and begins listening on a TCP port.
 func (s *Server) Start() {
 	router := mux.NewRouter()
```
Comment (arschles) [inline controller/server/server.go]: yep, good idea. added

Comment (krancour) [inline controller/server/server.go]: Thanks @arschles.

**Comment by krancour** [comment]
LGTM (again).

File: controller/server/server.go (hunk 1 above)
Comment (pmorie) [inline controller/server/server.go]: I strongly prefer issues over TODOs in the code personally.

**Comment by duglin** [comment]
LGTM

**Pull request merged**